pkg/util/cluster/cluster.go

```diff
@@ -104,6 +106,15 @@ func New(log *logrus.Entry, environment env.Core, ci bool) (*Cluster, error) {
 	scopes := []string{environment.Environment().ResourceManagerScope}
 	authorizer := azidext.NewTokenCredentialAdapter(spTokenCredential, scopes)
 
+	valutClient, err := keyvaultclient.NewVaultsClient(environment.SubscriptionID(), spTokenCredential, &policy.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: environment.Environment().Cloud,
+		},
```

Comment:
Sorry, I forgot to update my doc on this.  For the client options, let's use something like this or its equivalent instead of creating our own from scratch.

```
	environment.Environment().EnvironmentCredentialOptions()
```